frontend/src/components/worksheets/BundleDetail/BundleDetail.jsx

```diff
@@ -57,6 +59,18 @@ class BundleDetail extends React.Component<
         this.fetchBundleContents();
     }
 
+    componentDidUpdate() {
+        let newTime = new Date();
+        if (newTime.getTime() - this.state.lastFetchTime > 3000 && !this.state.fetching){
+            this.setState({fetching: true});
+            setTimeout(()=>{
+                this.setState({lastFetchTime: newTime.getTime(), fetching: false})
+                this.fetchBundleMetaData();

```

Comment:
Do we need to fetch both bundle metadata and bundle contents? Might it be more efficient to only fetch metadata, and then we might be able to tell if the contents could change?

Reply:
If you only fetch one of them, the other remains the same and won't be able to update, same as for fileBrower data